core/internal-groups.rst

```diff
@@ -0,0 +1,52 @@
+Internal Groups and Areas of Focus
+---------------------------------
+
+Core is probably the big component in coala. Almost everything else is heavily
+tied with it; so it doesn't make sense to have a huge, monolithic, and
+unfocussed team. So I propose to have a few areas of focus that each team member
+can choose to align themselves with.
+
+Please note that these are not strict teams. And by no means exhaustive. These
+is simply a categorization to co-ordinate efforts better. And it is entirely
+possible than an issue doesn't fall under one of the proposed groups squarely -
+overlaps are allowed. In short, these is simply a way to organize better, in an
+undivisive manner.
+
+Performance
+===========
+
+I'm a sucker for performance and I like things fast. And you should too!
+
+This group intends to improve on coala's performance. Since coala is written in
+Python, there are some limitations to how far we can go with this, but there's
+still quite a lot we can do.
+
+Usability and User Interface
+============================
+
+coala is designed to be very user-friendly and we should work to keep it that
+way. This means constant feedback, design and implementation. This also involves
+coming with new and intuitive interfaces. Just like fashion, UI development
+never ends.
+
+`bearlib` Development
+========================
+
+coala is nothing without its bears. This group will be dedicated to extending,
+improving, and maintaining the library available to bear authors.
+
+Supporting Tools
+================
+
+This was initially proposed as a separate team (informally). This group's main
+focus is the development of supporting tools for coala such as
+``coala-quickstart`` and ``coala-html``. This also involves taking care of
+internal libraries such as coala-utils.
+
+Implmenting New Ideas
```

Comment:
Hmm. I was just wondering if someone might want to implement already well-thought out ideas. For example, a full-fledged cEP for aspects was drafted and I implemented a lot of the core functionality - I didn't have much of a role in the design process.

>The research team is it's own team and members are not by default also core members

I'm sorry, I meant to say that "any core related research area will need to have an implementation in core". I agree that not every research member is a core member. I'll edit that.

>The research team also as other aspects besides core architecture. Mainly scientific work with coala.

I agree, I didn't fully realize the scope of the research team previously. Sorry about that.